source/MaterialXRender/LightHandler.cpp

```diff
@@ -75,9 +75,12 @@ void LightHandler::registerLights(DocumentPtr doc, const std::vector<NodePtr>& l
         }
     }
 
-    // Clamp the number of light sources to the number registered
-    unsigned int lightSourceCount = static_cast<unsigned int>(lights.size());
-    context.getOptions().hwMaxActiveLightSources = lightSourceCount;
+    // Make sure max light count is large enough
+    const unsigned int lightCount = (unsigned int)lights.size();
+    if (lightCount > context.getOptions().hwMaxActiveLightSources)
+    {
+        context.getOptions().hwMaxActiveLightSources = lightCount;
+    }
```

Comment:
Let the max count be a max limit, as it was intended, and only change it if we need to set it higher.